Content.Server/DeviceLinking/Systems/AlarmAutoLinkSystem.cs

```diff
@@ -0,0 +1,100 @@
+using Content.Server.Atmos.Components;
+using Content.Server.DeviceLinking.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Shared.Atmos;
+using Content.Shared.DeviceNetwork.Components;
+using Content.Shared.Doors.Components;
+using Content.Shared.Maps;
+using Robust.Server.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+
+namespace Content.Server.DeviceLinking.Systems;
+
+public sealed class AlarmAutoLinkSystem : EntitySystem
+{
+    [Dependency] private readonly DeviceListSystem _deviceListSystem = default!;
+    [Dependency] private readonly MapSystem _mapSystem = default!;
+    [Dependency] private readonly EntityLookupSystem _entityLookupSystem = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AlarmAutoLinkComponent, MapInitEvent>(OnMapInit);
+    }
+
+    private void OnMapInit(Entity<AlarmAutoLinkComponent> ent, ref MapInitEvent args)
+    {
+        if (!TryComp<DeviceListComponent>(ent, out var list))
+            return;
+
+        var xform = Transform(ent);
+        if (xform.GridUid is not { } gridUid || !TryComp(gridUid, out MapGridComponent? gridComp))
+            return;
+
+        var grid = new Entity<MapGridComponent>(gridUid, gridComp);
+        var indices = _mapSystem.TileIndicesFor(grid, xform.Coordinates);
+        // We indent one tile in the direction of rotation of the entity
+        var seed = (indices + xform.LocalRotation.ToWorldVec()).Floored();
+
+        var queue = new Queue<(TileRef, AtmosDirection)>();
+        var visited = new HashSet<(Vector2i, AtmosDirection)>();
+
+        queue.Enqueue((_mapSystem.GetTileRef(grid, seed), xform.LocalRotation.ToAtmosDirectionCardinal()));
+
+        var directions = new HashSet<(Vector2i, AtmosDirection)>
+        {
+            (Vector2i.Up, AtmosDirection.North),
+            (Vector2i.Down, AtmosDirection.South),
+            (Vector2i.Left, AtmosDirection.West),
+            (Vector2i.Right, AtmosDirection.East),
+        };
+
+        // Using flood-fill to iterate room tiles
+        while (queue.TryDequeue(out var node))
+        {
+            var (tileRef, direction) = node;
+            if (!visited.Add((tileRef.GridIndices, direction)))
+                continue;
+
+            var blockedDirections = AtmosDirection.Invalid;
+            var worldCoord = _mapSystem.LocalToWorld(grid.Owner, grid.Comp, tileRef.GridIndices);
+            var worldAABB = new Box2(worldCoord, worldCoord + grid.Comp.TileSizeVector).Enlarged(-0.05f);
+            var entities = _entityLookupSystem.GetEntitiesIntersecting(gridUid, worldAABB, LookupFlags.StaticSundries);
```

Comment:
```suggestion
            var blockedDirections = AtmosDirection.Invalid;
            var entities = _entityLookupSystem.GetLocalEntitiesInteracting(tileRef);
```

This code was not compensating for the rotation of the grid. It would sometimes consider solid walls in the first node and abort the flood fill search.

<img width="302" height="277" alt="image" src="https://github.com/user-attachments/assets/bbee6558-9af0-4c49-96a8-14b31996fb94" />